memguard_test.go

```diff
@@ -521,3 +505,42 @@ func TestGetBytes(t *testing.T) {
 		t.Error("pointer does not describe actual memory")
 	}
 }
+
+func TestFinalizer(t *testing.T) {
+	b, err := New(8, false)
+	if err != nil {
+		t.Error("unexpected error")
+	}
+	ib := b.lockedBuffer
+	// b is now unreachable
+
+	c, err := New(8, false)
+	if err != nil {
+		t.Error("unexpected error")
+	}
+	ic := c.lockedBuffer
+
+	if ib.IsDestroyed() != false {
```

Comment:
Is it not possible for this test to ever fail? What if there happens to be a GC run just before this check?